src/units/attack_type.cpp

```diff
@@ -135,7 +137,7 @@ static bool matches_simple_filter(const attack_type & attack, const config & fil
 	if(!filter_special.empty()) {
 		bool found = false;
 		for(auto& special : filter_special) {
-			if(attack.get_special_bool(special, true)) {
+			if(attack.get_special_bool(special, true, true, false)) {
```

Comment:
With this change, the meaning of existing WML code that uses filter_weapon with `special="foo"` will be changed. The language in https://wiki.wesnoth.org/FilterWML#Filtering_Weapons is unclear, but it seems reasonable that there's UMC out there that relies on `special=foo` matching either tag name or id. So, would it be preferable to keep the meaning of `special=foo` as-is, deprecate it, and add _two_ new keys, `special_id=foo` and `special_tag=foo` (as well as their `special_*_active` variants) instead?

Reply:
i will doing that then.

Reply:
Thanks! If you have any question, feel free to stop by the development channel on discord/IRC. See links here: https://github.com/wesnoth/wesnoth/blob/master/.github/CONTRIBUTING.md#contacting-us

Reply:
it is done

Reply:
Thanks! What's your opinion regarding whether or not to deprecate `"special"` and `"special_active"`?

Reply:
Well we can mark "special" as deprecate but keep it for version 1.16 as the beta version is imminent now and delete it only for the following versions to not take the UMC devellopeurs to the devoid

Reply:
> but keep it for version 1.16

Of course! That's the project policy for deprecations already: https://wiki.wesnoth.org/CompatibilityStandards

Okay, then let's deprecate both `special` and `special_active`. To do that, call [deprecated_message](https://github.com/wesnoth/wesnoth/blob/a0ccc2dff42996c3e2a0ba4f3fb19f259540dbe5/src/deprecation.hpp#L21-L44) if either of those keys is set. I suggest to deprecate at DEP_LEVEL::PREEMPTIVE.

Reply:
Don't forget to include deprecation.hpp